src/main/java/io/supertokens/totp/Totp.java

```diff
@@ -107,6 +106,11 @@ public static TOTPDevice registerDevice(AppIdentifierWithStorage appIdentifierWi
         }
 
         TOTPSQLStorage totpStorage = appIdentifierWithStorage.getTOTPStorage();
+        
+        if (deviceName == null) {
```

Comment:
If we are auto creating a new device, and the createDevice plugin interface function throws a DeviceAlreadyExists error, we need to do the following:
- If the existing device is verified, we need to retry with a new name (this should ideally be done via recursion).
- If the existing device is unverified, we need to replace the existing device with this new one (which is also unverified).